### PR TITLE
v_frame changes

### PR DESCRIPTION
**Pull request by master-of-zen**

### CODE REVIEW

File: v_frame/src/plane.rs [hunk 1]
```diff
@@ -837,6 +921,127 @@ pub mod test {
...
+        xpad: 0,
+        ypad: 0,
+        xorigin: 2,
+        yorigin: 3,
+      },
+    };
+    let downsampled = plane.downscale(2);
+
+    #[rustfmt::skip]
+    assert_eq!(
```
Comment (lu-zero) [inline v_frame/src/plane.rs:956]: You should compare the actual data and not the data + the padding.

File: v_frame/src/plane.rs [hunk 2]
```diff
@@ -837,6 +921,127 @@ pub mod test {
...
+        ypad: 0,
+        xorigin: 0,
+        yorigin: 0,
+      },
+    };
+
+    let downscaled = plane.downscale(3);
+
+    #[rustfmt::skip]
+    assert_eq!(
```
Comment (lu-zero) [inline v_frame/src/plane.rs:995]: Same here.

File: v_frame/src/plane.rs [hunk 3]
```diff
@@ -837,6 +921,127 @@ pub mod test {
...
+        xpad: 0,
+        ypad: 0,
+        xorigin: 0,
+        yorigin: 0,
+      },
+    };
+    let downsampled = plane.downscale(3);
+
+    #[rustfmt::skip]
+    assert_eq!(
```
Comment (lu-zero) [inline v_frame/src/plane.rs:1035]: And here.

File: v_frame/src/plane.rs [hunk 4]
```diff
@@ -473,6 +484,79 @@ impl<T: Pixel> Plane<T> {
...
+        let dst_rows = chunk.chunks_mut(stride);
+        for (row_offset, dst_row) in dst_rows.enumerate() {
+          let row_idx = chunk_idx * chunk_rows + row_offset;
+
+          // Iter dst cols
+          for (col_idx, dst) in dst_row[0..width].iter_mut().enumerate() {
+            let mut sum = half_box_pixels;
+
+            // Sum box of size scale * scale
+            // TODO: use SIMD here, maybe try `faster` crate
```
Comment (lu-zero) [inline v_frame/src/plane.rs:536]: Remove the reference to `faster` please.

**Review by lu-zero — CHANGES_REQUESTED** [review]
The tests are failing because you are comparing the padding, I'm afraid.

File: v_frame/src/plane.rs (hunk 1 above)
Comment (master-of-zen) [reply v_frame/src/plane.rs]: Fixed

File: v_frame/src/plane.rs (hunk 2 above)
Comment (master-of-zen) [reply v_frame/src/plane.rs]: Fixed

File: v_frame/src/plane.rs (hunk 3 above)
Comment (master-of-zen) [reply v_frame/src/plane.rs]: Fixed

File: v_frame/src/plane.rs (hunk 4 above)
Comment (master-of-zen) [reply v_frame/src/plane.rs]: Removed

**Review by lu-zero — APPROVED** [review]
Looks good, we'll have to decide if and when rename the tracing feature, when/if we'll switch to tracing to log information.

**Pull request merged**